apps/web/src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,63 @@
+import {
+  Listbox,
+  ListboxButton,
+  ListboxOptions,
+  Transition,
+} from "@headlessui/react";
+import { ChevronUpDownIcon } from "@heroicons/react/24/outline";
+
+import { Option } from "./components/Option";
+
+export interface Option {
+  value: string | number;
+  label: string;
+}
+
+export interface DropdownProps {
+  options: Option[];
+  selected: Option;
+  width?: string;
+  onChange(newOption: Option): void;
```

Comment:
`onChange` can be as follows:

`onChange(index, options)` where `index` is the position in the array of the selected element and `options` is props array

Reply:
I think the current solution is simpler and matches with most `selects/dropdowns` implementations along well-known libraries: when this event has been triggered, you should pass the new value of the HTML element. If you pass the `index` and the whole `options` array, you have two drawbacks:
- You will have to look for the value always when you change the value 
- You already have the `options` array when you pass it as a prop to the component so you would be sending redundant information

Reply:
Ok, let's continue with this approach